src/Symfony/Bundle/SecurityBundle/Resources/config/schema/security-1.0.xsd

```diff
@@ -84,6 +92,23 @@
         <xsd:attribute name="id" type="xsd:string" />
     </xsd:complexType>
 
+    <xsd:complexType name="password_hasher">
+        <xsd:sequence>
+            <xsd:element name="migrate-from" type="xsd:string" minOccurs="0" maxOccurs="unbounded" />
+        </xsd:sequence>
+        <xsd:attribute name="class" type="xsd:string" use="required" />
+        <xsd:attribute name="algorithm" type="xsd:string" />
```

Comment:
I guess there are tech reasons behind this, but it's confusing to see both `algorithm` and `hash-algorithm`

Reply:
I agree with you, this feels confusing. 

The main reason is that it has been introduced for the `pbkdf2`-based encoder, which is a key derivation function that requires to configure the underlying hash algorithm to be used (which can be any of `hash_algos()` as indicated by the SecurityBundle configuration).

We might want to rename it to `pbkdf2_algorithm` or equivalent. 
However, this PR is already quite big and full of renaming, I'd like to stop doing more changes. So I'll keep this issue for another PR before 5.3 if you don't mind.

Thank you!